src/EFCore/Metadata/Conventions/DerivedTypeDiscoveryConvention.cs

```diff
@@ -35,6 +35,7 @@ public DerivedTypeDiscoveryConvention([NotNull] ProviderConventionSetBuilderDepe
             var entityType = entityTypeBuilder.Metadata;
             var clrType = entityType.ClrType;
             if (clrType == null
+                || entityType.HasSharedClrType
                 || entityType.HasDefiningNavigation())
```

Comment:
```suggestion
                || entityType.HasSharedClrType
                || entityType.HasDefiningNavigation()
                || entityType.FindDeclaredOwnership() != null
                || entityType.Model.FindIsOwnedConfigurationSource(clrType) != null)
```

Should make this marginally faster on average